pkg/controller/kibana/pod.go

```diff
@@ -115,6 +129,51 @@ func GetKibanaContainer(podSpec corev1.PodSpec) *corev1.Container {
 	return pod.ContainerByName(podSpec, kbv1.KibanaContainerName)
 }
 
+func GetKibanaBasePathFromSpecEnv(podSpec corev1.PodSpec) string {
+	kbContainer := GetKibanaContainer(podSpec)
+	if kbContainer == nil {
+		return ""
+	}
+	for _, envVar := range kbContainer.Env {
+		if envVar.Name == "SERVER_BASEPATH" {
+			return envVar.Value
+		}
+	}
+	return ""
+}
+
 func getDefaultContainerPorts(kb kbv1.Kibana) []corev1.ContainerPort {
 	return []corev1.ContainerPort{{Name: kb.Spec.HTTP.Protocol(), ContainerPort: int32(network.HTTPPort), Protocol: corev1.ProtocolTCP}}
 }
+
+func getKibanaBasePath(kb kbv1.Kibana) (string, error) {
```

Comment:
In my testing and based on [the Kibana docs](https://www.elastic.co/guide/en/kibana/current/settings.html#server-rewriteBasePath) the basePath setting is only in effect and needs to be handled by ECK when combined with the `server.rewriteBasePath: true` option.

Reply:
Thanks, I missed this completely, but I encountered multiple redirect issues when I tried adding based path without this config in my testing, will add a check for this

Reply:
I made the change to look at rewriteBasePath